Tests/SDKTests.cs

```diff
@@ -0,0 +1,259 @@
+                                "            ],\n" +
+                                "            \"realm\": \"\",\n" +
+                                "            \"rights\": [\n" +
+                                "                \"policydelete\",\n" +
+                                "                \"resync\"\n" +
+                                "            ],\n" +
+                                "            \"role\": \"admin\",\n" +
+                                "            \"token\": \"" + authToken + "\",\n" +
+                                "            \"username\": \"admin\",\n" +
+                                "            \"logout_time\": 120,\n" +
+                                "            \"default_tokentype\": \"hotp\",\n" +
+                                "            \"user_details\": false,\n" +
+                                "            \"subscription_status\": 0\n" +
+                                "        }\n" +
+                                "    },\n" +
+                                "    \"time\": 1589446794.8502703,\n" +
+                                "    \"version\": \"privacyIDEA 3.2.1\",\n" +
+                                "    \"versionnumber\": \"3.2.1\",\n" +
+                                "    \"signature\": \"rsa_sha256_pss:\"\n" +
+                                "}"));
+
+            // Trigger challenge response
+            server
+                .Given(
+                    Request.Create()
+                    .WithPath("/validate/triggerchallenge")
+                    .UsingPost()
+                    .WithBody("user=test")
+                    .WithHeader("Authorization", authToken)
+                    .WithHeader("Content-Type", "application/x-www-form-urlencoded; charset=utf-8")
+                    )
+                .RespondWith(
+                    Response.Create()
+                    .WithStatusCode(200)
+                    .WithBody("{\n" +
+                                "  \"detail\": {\n" +
+                                "    \"attributes\": null,\n" +
+                                "    \"message\": \"Bitte geben Sie einen OTP-Wert ein: , Please confirm the authentication on your mobile device!\",\n" +
+                                "    \"messages\": [\n" +
+                                "      \"Bitte geben Sie einen OTP-Wert ein: \",\n" +
+                                "      \"Please confirm the authentication on your mobile device!\"\n" +
+                                "    ],\n" +
+                                "    \"multi_challenge\": [\n" +
+                                "      {\n" +
+                                "        \"attributes\": null,\n" +
+                                "        \"message\": \"Bitte geben Sie einen OTP-Wert ein: \",\n" +
+                                "        \"serial\": \"OATH00020121\",\n" +
+                                "        \"transaction_id\": \"02659936574063359702\",\n" +
```

Comment:
You can already add the `client_mode` in the response here.

Reply:
I dont think it makes sense to add this here prematurely when it is not processed by the code yet.

Reply:
OK.